bignumber.d.ts

```diff
@@ -330,10 +330,28 @@ export namespace BigNumber {
     suffix?: string;
   }
 
+  export interface BigNumberData {
+    /**
+     * The coefficient of the value of this BigNumber, an array of base 1e14 integer numbers.
+     */
+    readonly c: number[];
+
+    /**
+     * The exponent of the value of this BigNumber, an integer number, -1000000000 to 1000000000.
+     */
+    readonly e: number;
+
+    /**
+     * The sign of the value of this BigNumber, -1 or 1.
+     */
+    readonly s: number;
+
+    [key: string]: any;
```

Comment:
Does this stop the object having a symbol as a key? If so, that would be a [problem](https://github.com/MikeMcl/bignumber.js/blob/master/bignumber.mjs#L2695).

Reply:
according to TS1023
```
An index signature parameter type must be 'string' or 'number'.  ts(1023)
```

It looks like [it's ok](https://www.typescriptlang.org/play/index.html#src=interface%20IInterface%20%7B%0D%0A%20%20num%3A%20number%2C%0D%0A%20%20%5Bkey%3A%20string%5D%3A%20any%0D%0A%7D%0D%0A%0D%0Aconst%20x%3A%20IInterface%20%3D%20%7B%0D%0A%20%20%20%20num%3A%2012%2C%0D%0A%20%20%20%20x%3A%20'string'%2C%0D%0A%20%20%20%20%5BSymbol.toStringTag%5D%3A%20'BigNumber'%0D%0A%7D%0D%0A%0D%0Aconsole.log(x)%0D%0A%0D%0Aconst%20keys%20%3D%20Object.keys(x)%0D%0Aconsole.log(keys%5B0%5D)).

Reply:
Looks like it's still an [issue](https://github.com/Microsoft/TypeScript/issues/1863), so probably best without it.

Reply:
but then it won't work with old BigNumber versions (when `instanceof` doesn't work).

not sure I understand, will you rewrite BigNumber in typescript? 
what I wanted was to allow unknown props, as it will be much easier to construct BigNumber from it. 

for example in js you will be able to do
```javascript
new BigNumber({s: null, e: null, c: null, userSpecific: 'v10' })
```

but in typescript, if we remove the `[key: string]: any`:

```typescript
new BigNumber({s: null, e: null, c: null, userSpecific: 'v10' } as BigNumber.Object)
```

Reply:
>> but then it won't work with old BigNumber versions (when `instanceof` doesn't work).

Sorry, I'm not following you. Why would it not work?  (Incidentally, old BigNumber versions have always worked: the constructor just calls `toString` on them.)